src/prototype_creep_routing.js

```diff
@@ -15,38 +15,9 @@ Creep.prototype.getRoute = function() {
   let route = [];
   let creep = this;
   if (this.memory.base != this.memory.routing.targetRoom) {
-    // TODO more dynamic, room.memory.hostile value?
-    let routeCallback = function(roomName, fromRoomName) {
-      if (roomName == creep.memory.routing.targetRoom) {
-        return 1;
-      }
-
-      if (Memory.rooms[roomName] && Memory.rooms[roomName].state == 'Occupied') {
-        //         console.log(`Creep.prototype.getRoute: Do not route through occupied rooms ${roomName}`);
-        if (config.path.allowRoutingThroughFriendRooms && friends.indexOf(Memory.rooms[roomName].player) > -1) {
-          console.log('routing through friendly room' + roomName);
-          return 1;
-        }
-        //         console.log('Not routing through enemy room' + roomName);
-        return Infinity;
-      }
-
-      if (Memory.rooms[roomName] && Memory.rooms[roomName].state == 'Blocked') {
-        //         console.log(`Creep.prototype.getRoute: Do not route through blocked rooms ${roomName}`);
-        return Infinity;
-      }
-
-      return 1;
-    };
-    route = Game.map.findRoute(this.memory.base, this.memory.routing.targetRoom, {
-      routeCallback: routeCallback
-    });
-    //     this.log('getRoute: ' + this.memory.base + ' target: ' + this.memory.routing.targetRoom + ' route: ' + route);
-    // TODO I guess some bug while transitionen to memory.routing
-    if (route == -2) {
-      route = [];
-    }
+    route = this.room.findRoute(this.memory.base, this.memory.routing.targetRoom);
```

Comment:
We always calculate the routes from `base` to `targetRoom`. In an older version I used `this.room.name`, but I ran into different kind of problems.

If we have cases where we want to have it different, we can discuss. For now I would keep it as it is.